signature/src/agent.rs

```diff
@@ -91,14 +105,29 @@ impl Agent {
     /// Then save the gathered data into `path`
     async fn get_resource(&mut self, resource_name: &str, path: &str) -> Result<()> {
         let resource_description = serde_json::to_string(&ResourceDescription::new(resource_name))?;
-        let req = tonic::Request::new(GetResourceRequest {
-            kbc_name: self.kbc_name.clone(),
-            kbs_uri: self.kbc_uri.clone(),
-            resource_description,
-        });
-        let res = self.client.get_resource(req).await?;
-
-        fs::write(path, res.into_inner().resource).await?;
+        let res = if self.enable_native_aa {
+            AttestationAgent::new()
+                .download_confidential_resource(
```

Comment:
A quick question: 
every time we call `get_resource()` will create a new `AttestationAgent` object. What if we keep the `AA` object as a member of `Agent`, s.t.
```rust
pub struct Agent {
    /// Get Resource Service gRPC client
    client: Option<GetResourceServiceClient<Channel>>,
    aa: Option<AttestationAgent>,
    /// Enable Get Rserouce native AA client
    enable_native_aa: bool,
    kbc_name: String,
    kbs_uri: String,
}
```

Or an enum for `GetResourceServiceClient<Channel>` and `AttestationAgent`, for extensibility., i.e.
```rust
enum Client {
    GRPC(GetResourceServiceClient<Channel>),
    AA(AttestationAgent),
}
pub struct Agent {
    /// Get Resource Service gRPC client
    client: Client,
    enable_native_aa: bool,
    kbc_name: String,
    kbs_uri: String,
}
```

In both ways every time we call `get_resource()` will not do duplicated thins as creations and dropping of the aa object. How do you think about this?

Reply:
Sure. We choose to use the enum to wrap the real client. Thanks for your great comments!